pythonPackages: fix some broken builds

xeji opened the pull request:
###### Motivation for this change

ZHF #45960. Fix various broken python packages.

###### Things done

- [x] Built in sandbox on NixOS x86_64-linux
- [x] passes `nix-review`

---

xeji committed:
pythonPackages.daphne: 2.1.0 -> 2.2.2

Update, re-enable all tests on linux, disable tests on darwin.
This fixes the build of pythonPackages.channels

xeji commented:
@GrahamcOfBorg build python36Packages.daphne python36Packages.channels

GrahamcOfBorg commented:
<!--REQUEST_ID=5fd1cf77-28b7-4e70-a90e-8d4f050f8dcd-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=c889ba7f-f18a-4436-a65c-c91e8df40e39)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed channels-2.1.3
/build/channels-2.1.3
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3/lib
patching script interpreter paths in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
checking for references to /build in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3...
/nix/store/dfkz790ci59y942a7s5377bx8a9kn127-python3.6-daphne-2.2.2
/nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=5fd1cf77-28b7-4e70-a90e-8d4f050f8dcd-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=739b9cf4-02ad-43d7-a3e5-b281a80b2ab8)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed channels-2.1.3
/build/channels-2.1.3
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3/lib
patching script interpreter paths in /nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3
checking for references to /build in /nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3...
/nix/store/v214m4igqww33hvyj8nfhw7z8zzidh8r-python3.6-daphne-2.2.2
/nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=5fd1cf77-28b7-4e70-a90e-8d4f050f8dcd-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=e70e4453-3a33-4e73-a57f-13be2bc03745)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```

tests/test_websocket.py::TestWebsocket::test_http_bits
  /nix/store/3hzkfm9cmyp1g7j82vghl8gyfs7qhyg8-python3.6-hypothesis-3.66.2/lib/python3.6/site-packages/hypothesis/strategies.py:588: HypothesisDeprecationWarning: You should remove the average_size argument, because it is deprecated and no longer has any effect.  Please open an issue if the default distribution of examples does not work for you.
    check_valid_sizes(min_size, average_size, max_size)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============== 27 failed, 21 passed, 9 warnings in 149.82 seconds ==============
builder for '/nix/store/k5934q5ck2q9iqhlf1z2rsq6rnc52pp7-python3.6-daphne-2.2.2.drv' failed with exit code 1
cannot build derivation '/nix/store/f7z19gxchfqfk0jp73dgy2yqqynq6xqb-python3.6-channels-2.1.3.drv': 1 dependencies couldn't be built
error: build of '/nix/store/f7z19gxchfqfk0jp73dgy2yqqynq6xqb-python3.6-channels-2.1.3.drv', '/nix/store/k5934q5ck2q9iqhlf1z2rsq6rnc52pp7-python3.6-daphne-2.2.2.drv' failed
```
</p></details>

xeji commented:
@GrahamcOfBorg build python36Packages.daphne python36Packages.channels

GrahamcOfBorg commented:
<!--REQUEST_ID=1d7e7044-8828-4068-86bc-b5952641d42e-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=838f71f8-955a-45cf-be04-46da5c4ff359)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/v214m4igqww33hvyj8nfhw7z8zzidh8r-python3.6-daphne-2.2.2
/nix/store/wv6z181m124n1hh69yf8p28azpmkra0s-python3.6-channels-2.1.3
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=1d7e7044-8828-4068-86bc-b5952641d42e-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=08470ddb-bafb-4b0d-b097-729cbdcc5f2c)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```
Requirement already satisfied: idna>=2.5 in /nix/store/6d5iv9w1k93gcnd2aw3dl7z9gkkga9c4-python3.6-idna-2.7/lib/python3.6/site-packages (from hyperlink>=17.1.1->twisted>=18.7->daphne~=2.2->channels==2.1.3) (2.7)
Installing collected packages: channels
Successfully installed channels-2.1.3
/private/tmp/nix-build-python3.6-channels-2.1.3.drv-0/channels-2.1.3
post-installation fixup
strip is /nix/store/df6k4mgdjxciy0f637lryp7c9ln7n1m3-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/6zw02k6hd3nmrn5r1p7by9ifdzxd65fv-python3.6-channels-2.1.3/lib
patching script interpreter paths in /nix/store/6zw02k6hd3nmrn5r1p7by9ifdzxd65fv-python3.6-channels-2.1.3
/nix/store/ng0qx7gg2sj6iq11mlv8rnl877fdcl62-python3.6-daphne-2.2.2
/nix/store/6zw02k6hd3nmrn5r1p7by9ifdzxd65fv-python3.6-channels-2.1.3
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=1d7e7044-8828-4068-86bc-b5952641d42e-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=09d3dd90-55cb-4ae9-bf64-b1ecd77168f8)

Attempted: python36Packages.daphne, python36Packages.channels

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed channels-2.1.3
/build/channels-2.1.3
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3/lib
patching script interpreter paths in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
checking for references to /build in /nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3...
/nix/store/dfkz790ci59y942a7s5377bx8a9kn127-python3.6-daphne-2.2.2
/nix/store/glbhvpkbgpnb8cdi0rv9140g0vmkyc68-python3.6-channels-2.1.3
```
</p></details>

xeji committed:
pythonPackages.eve: fix build, drop incorrect dependencies

Build failed because of dependency Flask-PyMongo, which actually
isn't a dependency of eve as they have their own flask_pymongo code.

- Drop incorrect dependency flask-pymongo
- Drop redunant dependencies already propageted by flask:
  itsdangerous, werkzeug, jinja2
- Drop dependency markupsafe, not required any more

xeji commented:
@GrahamcOfBorg build python27Packages.eve python36Packages.eve

GrahamcOfBorg commented:
<!--REQUEST_ID=8cb29a0c-9f24-43f1-991b-36045a81d024-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=8af36d4f-5781-4d16-abdf-e27115e1679a)

Attempted: python27Packages.eve, python36Packages.eve

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed Eve-0.8
/build/Eve-0.8
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/5gxdjr4gwpz6ll3ycs5600b5ym9z6gbr-python3.6-Eve-0.8
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/5gxdjr4gwpz6ll3ycs5600b5ym9z6gbr-python3.6-Eve-0.8/lib
patching script interpreter paths in /nix/store/5gxdjr4gwpz6ll3ycs5600b5ym9z6gbr-python3.6-Eve-0.8
checking for references to /build in /nix/store/5gxdjr4gwpz6ll3ycs5600b5ym9z6gbr-python3.6-Eve-0.8...
/nix/store/j3jhzr2fc7f12lq1akxb5cammz5652qj-python2.7-Eve-0.8
/nix/store/5gxdjr4gwpz6ll3ycs5600b5ym9z6gbr-python3.6-Eve-0.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8cb29a0c-9f24-43f1-991b-36045a81d024-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=3f199404-ff27-4637-9b55-2d963b1730a4)

Attempted: python27Packages.eve, python36Packages.eve

<details><summary>Partial log (click to expand)</summary><p>

```
Requirement already satisfied: MarkupSafe>=0.23 in /nix/store/5f53wy70y3ifxymjhpf6r9891qksijhp-python3.6-markupsafe-1.0/lib/python3.6/site-packages (from Jinja2>=2.10->flask>=1.0->Eve==0.8) (1.0)
Installing collected packages: Eve
Successfully installed Eve-0.8
/private/tmp/nix-build-python3.6-Eve-0.8.drv-0/Eve-0.8
post-installation fixup
strip is /nix/store/df6k4mgdjxciy0f637lryp7c9ln7n1m3-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/4qrdvangn8sl6pj50y9iijglb0d8nq3m-python3.6-Eve-0.8/lib
patching script interpreter paths in /nix/store/4qrdvangn8sl6pj50y9iijglb0d8nq3m-python3.6-Eve-0.8
/nix/store/3fhsz6njf1hpx9h17lh9x4m4yzlyxm14-python2.7-Eve-0.8
/nix/store/4qrdvangn8sl6pj50y9iijglb0d8nq3m-python3.6-Eve-0.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8cb29a0c-9f24-43f1-991b-36045a81d024-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=4db50c7f-c787-4ba9-899e-c69e14271033)

Attempted: python27Packages.eve, python36Packages.eve

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed Eve-0.8
/build/Eve-0.8
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/kcgvyjxk848fni8x7q85g8r6ag76v64v-python3.6-Eve-0.8
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/kcgvyjxk848fni8x7q85g8r6ag76v64v-python3.6-Eve-0.8/lib
patching script interpreter paths in /nix/store/kcgvyjxk848fni8x7q85g8r6ag76v64v-python3.6-Eve-0.8
checking for references to /build in /nix/store/kcgvyjxk848fni8x7q85g8r6ag76v64v-python3.6-Eve-0.8...
/nix/store/jd1qiqk1q3f5jdvra3kw673mnsxpq6w6-python2.7-Eve-0.8
/nix/store/kcgvyjxk848fni8x7q85g8r6ag76v64v-python3.6-Eve-0.8
```
</p></details>

xeji committed:
pythonPackages.flask-pymongo: fix build

add missing dependency: vcversioner

xeji committed:
pythonPackages.pytest-rerunfailures: fix build

some tests fail since pytest 3.7.2 -> 3.7.4 update, disable them

xeji committed:
pythonPackages.wordfreq: 2.0 -> 2.2.0, fix build

Previous version didn't build (test failure).
Update to latest, tests use pytest now, disable failing tests.

xeji commented:
@GrahamcOfBorg build python27Packages.flask-pymongo python36Packages.flask-pymongo  python27Packages.pytest-rerunfailures python36Packages.pytest-rerunfailures python36Packages.wordfreq

GrahamcOfBorg commented:
<!--REQUEST_ID=743d9016-48e6-43a1-ba73-0e9925afc507-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=1c3429cc-3b3d-4d21-abfb-179e9b598803)

Attempted: python27Packages.flask-pymongo, python36Packages.flask-pymongo, python27Packages.pytest-rerunfailures, python36Packages.pytest-rerunfailures, python36Packages.wordfreq

<details><summary>Partial log (click to expand)</summary><p>

```
tests/test_chinese.py ...                                                [ 25%]
tests/test_general.py ...............                                    [ 87%]
tests/test_transliteration.py ...                                        [100%]

=================== 24 passed, 9 deselected in 4.71 seconds ====================
/nix/store/pl6m9s98k9giblsa2b46g53jvaczyfbj-python2.7-Flask-PyMongo-2.1.0
/nix/store/zfww8j0146a7qhav7grkjp7yr096dan1-python3.6-Flask-PyMongo-2.1.0
/nix/store/yq9zxh5w6dm23zdqrqq8bzwcabfh6iz3-python2.7-pytest-rerunfailures-4.1
/nix/store/rr2xvjbfzh48a1zskyy8nax3gpy2f6fv-python3.6-pytest-rerunfailures-4.1
/nix/store/0laqakjzzvycxv38kgvl9anpc4nnb668-python3.6-wordfreq-2.2.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=743d9016-48e6-43a1-ba73-0e9925afc507-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=84eb3f0a-2e1c-4397-897e-7776393b04e9)

Attempted: python27Packages.flask-pymongo, python36Packages.flask-pymongo, python27Packages.pytest-rerunfailures, python36Packages.pytest-rerunfailures, python36Packages.wordfreq

<details><summary>Partial log (click to expand)</summary><p>

```
tests/test_chinese.py ...                                                [ 25%]
tests/test_general.py ...............                                    [ 87%]
tests/test_transliteration.py ...                                        [100%]

=================== 24 passed, 9 deselected in 4.96 seconds ====================
/nix/store/gnz1h3r5gdzd6lz4j56p3yabbhqfb6n0-python2.7-Flask-PyMongo-2.1.0
/nix/store/ds6ijl3ch1wgnshpqfzfgawlrp4x3gra-python3.6-Flask-PyMongo-2.1.0
/nix/store/nx9qz4p59kd2sxn9d5krr89plwdkcssv-python2.7-pytest-rerunfailures-4.1
/nix/store/jrlf4cz773wnylsfw708gb5vk67rd4mk-python3.6-pytest-rerunfailures-4.1
/nix/store/yyh5pq90w8abld3nhff653df4db8vajn-python3.6-wordfreq-2.2.0
```
</p></details>

xeji committed:
pythonPackages.nilearn: fix tests

disable failing tests

xeji commented:
@GrahamcOfBorg build python27Packages.nilearn python36Packages.nilearn

GrahamcOfBorg commented:
<!--REQUEST_ID=5654dc28-16a8-418d-8a56-ea8a8b8556a5-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=d899a7a7-9a11-4af7-b7c3-1667f27f607d)

Attempted: python27Packages.nilearn, python36Packages.nilearn

<details><summary>Partial log (click to expand)</summary><p>

```
nilearn/input_data/nifti_spheres_masker.py     116     92    21%
nilearn/masking.py                             196     39    80%
nilearn/signal.py                              169     55    67%
nilearn/version.py                              25      8    68%
----------------------------------------------------------------
TOTAL                                         4009   2433    39%
----------------------------------------------------------------------
Ran 161 tests in 6.086s

OK
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=5654dc28-16a8-418d-8a56-ea8a8b8556a5-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=0a341907-3a10-482f-bcd6-c34417c052dd)

Attempted: python27Packages.nilearn, python36Packages.nilearn

<details><summary>Partial log (click to expand)</summary><p>

```
nilearn/signal.py                              169     55    67%
nilearn/version.py                              25      8    68%
----------------------------------------------------------------
TOTAL                                         4009   2433    39%
----------------------------------------------------------------------
Ran 161 tests in 7.556s

OK
/nix/store/wrzignqy34k5nhcafibf9ag7v2dqqgcm-python2.7-nilearn-0.4.2
/nix/store/3g92356qfqll5cfqmwm5yv1qf7syv9nm-python3.6-nilearn-0.4.2
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=5654dc28-16a8-418d-8a56-ea8a8b8556a5-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46608&attempt_id=f0b451aa-1121-42c4-97ce-d9b640426001)

Attempted: python27Packages.nilearn, python36Packages.nilearn

<details><summary>Partial log (click to expand)</summary><p>

```
nilearn/signal.py                              169     55    67%
nilearn/version.py                              25      8    68%
----------------------------------------------------------------
TOTAL                                         4009   2433    39%
----------------------------------------------------------------------
Ran 161 tests in 61.700s

OK
/nix/store/fgabfnnl13h0p5fa2n1h4vzsldp1h3qn-python2.7-nilearn-0.4.2
/nix/store/3f4pl6q66l906irpzac21d3f54dx0isc-python3.6-nilearn-0.4.2
```
</p></details>

xeji commented:
backported: 8a93d258cd8..5eb4258a9a1
